rust/processor/src/models/fungible_asset_models/v2_fungible_metadata.rs

```diff
@@ -135,55 +132,19 @@ impl FungibleAssetMetadataModel {
 
     /// A fungible asset can also be a token. We will make a best effort guess at whether this is a fungible token.
     /// 1. If metadata is present without token object, then it's not a token
-    /// 2. If metadata is not present, we will do a lookup in the db. If it's not there, it's a token
+    /// 2. If metadata is not present, we will do a lookup in the db.
     pub async fn is_address_fungible_asset(
         conn: &mut PgPoolConnection<'_>,
         address: &str,
         fungible_asset_metadata: &ObjectAggregatedDataMapping,
+        txn_version: i64,
     ) -> bool {
         if let Some(metadata) = fungible_asset_metadata.get(address) {
             metadata.token.is_none()
         } else {
-            // Look up in the db
-            Self::query_is_address_fungible_asset(conn, address).await
-        }
-    }
-
-    /// Try to see if an address is a fungible asset (not a token). We'll try a few times in case there is a race condition,
-    /// and if we can't find after N times, we'll assume that it's not a fungible asset.
-    /// TODO: An improvement is to combine this with is_address_token. To do this well we need
-    /// a k-v store
-    async fn query_is_address_fungible_asset(
-        conn: &mut PgPoolConnection<'_>,
-        address: &str,
-    ) -> bool {
-        let mut retried = 0;
-        while retried < QUERY_RETRIES {
-            retried += 1;
-            match Self::get_by_asset_type(conn, address).await {
-                Ok(_) => return true,
-                Err(_) => {
-                    std::thread::sleep(std::time::Duration::from_millis(QUERY_RETRY_DELAY_MS));
-                },
-            }
+            // Look up in the db. The object must exist in current_objects table for this processor to proceed
+            let object = Object::get_object(conn, address, txn_version).await;
+            object.is_fungible_asset && !object.is_token
```

Comment:
These should be nullable so...my guess is that you want to keep the old code for now in case these are nullable, but add an warning that they should backfill the object processor.